### PR TITLE
LPD-32130 Fix current user id value in payload information

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://liferay.atlassian.net/browse/LPD-32130

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java [hunk 1]
```diff
@@ -232,7 +232,7 @@ private static Map<String, Object> _getVariables(
 					return dateFormat.format(new Date());
 				}
 			).put(
-				"currentUserId", payloadJSONObject.getString("userId")
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java:235]: @marcelabc @icgsbr this code was introduced to fix https://liferay.atlassian.net/browse/LPS-185225. 

Is it ok to change this?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java [hunk 2]
```diff
@@ -419,6 +430,35 @@ public void testAddObjectValidationRule() throws Exception {
...
+
+		Assert.assertEquals(1, _argumentsList.size());
+
+		Object[] arguments = _argumentsList.poll();
+
+		Map<String, Object> inputObjects = (Map<String, Object>)arguments[0];
+
+		Assert.assertEquals(
+			TestPropsValues.getUserId(), inputObjects.get("currentUserId"));
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java:461]: @marcelabc @icgsbr It's better to use try/finally here:

```
ObjectScriptingExecutor originalObjectScriptingExecutor =
	(ObjectScriptingExecutor)_getAndSetFieldValue(
		ObjectScriptingExecutor.class, "_objectScriptingExecutor",
		ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY);

try {
	Assert.assertEquals(0, _argumentsList.size());

	_objectEntryLocalService.addObjectEntry(
		TestPropsValues.getUserId(), 0,
		_objectDefinition.getObjectDefinitionId(),
		HashMapBuilder.<String, Serializable>put(
			"textObjectField", RandomTestUtil.randomString()
		).build(),
		ServiceContextTestUtil.getServiceContext());

	Assert.assertEquals(1, _argumentsList.size());

	Object[] arguments = _argumentsList.poll();

	Map<String, Object> inputObjects = (Map<String, Object>)arguments[0];

	Assert.assertEquals(
		TestPropsValues.getUserId(), inputObjects.get("currentUserId"));
}
finally {
	ReflectionTestUtil.setFieldValue(
		_objectValidationRuleEngineRegistry.getObjectValidationRuleEngine(
			0, ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY),
		"_objectScriptingExecutor", originalObjectScriptingExecutor);
}
```

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java [hunk 3]
```diff
@@ -419,6 +430,35 @@ public void testAddObjectValidationRule() throws Exception {
...
+			"_objectScriptingExecutor", originalObjectScriptingExecutor);
+
+		Assert.assertEquals(1, _argumentsList.size());
+
+		Object[] arguments = _argumentsList.poll();
+
+		Map<String, Object> inputObjects = (Map<String, Object>)arguments[0];
+
+		Assert.assertEquals(
+			TestPropsValues.getUserId(), inputObjects.get("currentUserId"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java:460]: @marcelabc @icgsbr we can try to use the MapUtil here:

```
MapUtil.getLong((Map<String, Object>)arguments[0], "currentUserId")
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java (hunk 1 above)
Comment (marcelabc) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/util/ObjectEntryVariablesUtil.java:235]: Thank you!
I added a fix so we don't make the same regression.

**Comment by marcelabc** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Pull request closed without merging**